numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -468,51 +470,79 @@ def __getitem__(self, args):
             global_range and local_range attributes initialized.
 
         """
-        if isinstance(args, int):
-            self._global_range = [args]
-            self._local_range = None
-        elif isinstance(args, tuple) or isinstance(args, list):
-            if len(args) == 1 and all(isinstance(v, int) for v in args):
-                self._global_range = list(args)
-                self._local_range = None
-            elif len(args) == 2:
-                gr = args[0]
-                lr = args[1]
-                if isinstance(gr, int):
-                    self._global_range = [gr]
-                elif len(gr) != 0 and all(isinstance(v, int) for v in gr):
-                    self._global_range = list(gr)
-                else:
-                    raise IllegalRangeValueError(kernel_name=self.kernel_name)
+        warn(
+            """
+            The current syntax for specification of kernel lauch parameters
+            will be deprecated soon. Users are requested to do it through
+            Range/NdRange classes.
+                Example:
+                    from numba_dpex.core.kernel_interface.utils import NdRange
+                    ranges = NdRange((8,8), (2,2)) # 2D global and local ranges
+                    <function>[ranges](<parameters>)
+            """,
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        if isinstance(args, NdRange):
```

Comment:
The change does not match SYCL semantics. The local range for an NdRange is not optional. We really need two classes `Range` and `NdRange`. There is a difference in how a kernel is submitted based on  whether a range or an ndrange is provided. 

We have to be clear and explicit about the usage. When using NdRange class it should without ambiguity indicate that the kernel will be submitted as an ndrange kernel.

Reply:
@diptorupd Right now, the correct syntax for kernel parameters are like one of these:

```python
ranges = NdRange((2,2,2),(1,1,1))
func[ranges](a,b,c)
```

or 

```python
ranges = NdRange((2,2,2))
func[ranges](a,b,c)
```

or

```python
global_range = Range(2,2,2)
local_range = Range(1,1,1)
ranges = NdRange(global_range, local_range)
func[ranges](a,b,c)
```

or

```python
global_range = Range(2,2,2)
ranges = NdRange(global_range)
func[ranges](a,b,c)
```

If these are not correct, could you give me an example of a correct syntax?

Reply:
In the above examples, (1) and (3) is what we should be support.
(2) and (4) we should not.

That is, `NdRange` should require both `global_range` and `local_range` to be set. We should not allow `NdRange` where `local_range` is `None`.

Reply:
> We should not allow `NdRange` where `local_range` is `None`.

But there are many examples where we don't know the `local_range` and therefore we use `numba_dpex.DEFAULT_LOCAL_SIZE` (which is basically a `None`). For example, what do we do in the `sum2D.py` in the `examples` folder?

Reply:
Fixed.